### PR TITLE
GUI Test 2

### LINKED ISSUE #174: GUI Testing
- [ ] Switch View
- [ ] Help View (count how many are there)
- [ ] Find Result (the top bar)
- [ ] Global Tag View
- [ ] Error View (Test presence)

### PR DESCRIPTION
**Pull request by xiendong**
Resolves #174

### CODE REVIEW

File: src/main/java/seedu/todo/logic/commands/CommandMap.java [hunk 1]
```diff
@@ -86,4 +87,16 @@ private static void buildCommandSummariesMap() {
...
         return commandSummariesList;
     }
+
+    //@@author A0135805H
+    /**
+     * Get a representation of the command summary by a list of array strings, with each array string
+     * with the format:
+     *      [Scenario, Command, Arguments]
+     */
+    public static List<String[]> getAllCommandSummaryArray() {
```
Comment (li-kai) [inline src/main/java/seedu/todo/logic/commands/CommandMap.java]: Use `getAllCommandSummary` which encapsulates the three fields already.

File: src/test/java/guitests/guihandles/FindCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        tasks.add(TaskBuilder.name("Kaboom")
+                .due(LocalDateTime.now().plusDays(1))
+                .tagged("dynamite", "pikachu")
+                .build());
+
+        return getInitialDataHelper(tasks);
+    }
+
+    @Test
+    public void find_NoResults() {
```
Comment (li-kai) [inline src/test/java/guitests/guihandles/FindCommandTest.java]: `find_noResults`

File: src/test/java/guitests/guihandles/HelpViewHandle.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        }
+        return displayedHelpItems;
+    }
+
+    /**
+     * Returns true if this help view displays all the help items correctly, by checking if the data provided
+     * by {@code commandSummaries} matches to what is displayed.
+     */
+    public boolean isHelpItemsDisplayedCorrectly(List<String[]> commandSummaries) {
+        List<String[]> displayedCommandSummaries = getDisplayedHelpItems();
```
Comment (li-kai) [inline src/test/java/guitests/guihandles/HelpViewHandle.java:80]: Might be more troublesome for you here. Sorry though, its not a good practice to add production code just to test it. 😄

File: src/test/java/guitests/guihandles/SearchStatusViewHandle.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+     */
+    public String getSearchTermText() {
+        Text text = (Text) getNode(SEARCH_TERM_LABEL_ID);
+        return text.getText();
+    }
+
+    /**
+     * Returns true if the number of items found matches the count displayed in this view.
+     */
+    public boolean doesSearchCountMatch(int count) {
```
Comment (li-kai) [inline src/test/java/guitests/guihandles/SearchStatusViewHandle.java:51]: You're testing implementation here. Just check if text is present or 'task' is found. If you do want to make sure it works, write a unit test for it, but not for GUI test.

File: src/test/java/seedu/todo/testutil/CollectionUtil.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+ * Utility methods related to Collections
+ */
+public class CollectionUtil {
+    /**
+     * Returns true if every element in a collection are unique by {@link Object#equals(Object)}.
+     */
+    public static boolean elementsAreUnique(Collection<?> items) {
+        final Set<Object> testSet = new HashSet<>();
```
Comment (li-kai) [inline src/test/java/seedu/todo/testutil/CollectionUtil.java]: There's an easier way. Just convert the collection to a set and check if the length are the same! 😆

**Review by li-kai — CHANGES_REQUESTED** [review]
Looks good. Just a few changes.

File: src/main/java/seedu/todo/logic/commands/CommandMap.java (hunk 1 above)
Comment (xiendong) [reply src/main/java/seedu/todo/logic/commands/CommandMap.java]: I want them in array of strings :/

Comment (xiendong) [reply src/main/java/seedu/todo/logic/commands/CommandMap.java]: Okay bah I shifted this method back to `HelpCommandTest`

File: src/test/java/guitests/guihandles/FindCommandTest.java (hunk 2 above)
Comment (xiendong) [reply src/test/java/guitests/guihandles/FindCommandTest.java]: oops

File: src/test/java/guitests/guihandles/HelpViewHandle.java (hunk 3 above)
Comment (xiendong) [reply src/test/java/guitests/guihandles/HelpViewHandle.java:80]: Which production code are you referring to?

File: src/test/java/guitests/guihandles/SearchStatusViewHandle.java (hunk 4 above)
Comment (xiendong) [reply src/test/java/guitests/guihandles/SearchStatusViewHandle.java:51]: This just checks if the number is displayed correctly in the `SearchStatusView` though, nothing else.

File: src/test/java/seedu/todo/testutil/CollectionUtil.java (hunk 5 above)
Comment (xiendong) [reply src/test/java/seedu/todo/testutil/CollectionUtil.java]: This is given code :p

Comment (xiendong) [reply src/test/java/seedu/todo/testutil/CollectionUtil.java]: Updated

**Pull request merged**